explorer-api/src/geo_ip/location.rs

```diff
@@ -78,20 +99,30 @@ impl GeoIp {
                 &address
             );
             let p = port.unwrap_or(FAKE_PORT);
-            let socket_addr = (address, p)
-                .to_socket_addrs()
-                .map_err(|e| {
-                    error!("Fail to resolve IP address from {}:{}: {}", &address, p, e);
-                    GeoIpError::NoValidIP
-                })?
-                .next()
-                .ok_or_else(|| {
-                    error!("Fail to resolve IP address from {}:{}", &address, p);
-                    GeoIpError::NoValidIP
-                })?;
-            let ip = socket_addr.ip();
-            debug!("Internal lookup succeed, resolved ip: {}", ip);
-            Ok(ip)
+            match (address, p).to_socket_addrs() {
+                Ok(mut addrs) => {
+                    if let Some(socket_addr) = addrs.next() {
+                        let ip = socket_addr.ip();
+                        debug!("Internal lookup succeeded, resolved ip: {}", ip);
+                        Ok(ip)
+                    } else {
+                        debug!("Fail to resolve IP address from {}:{}", &address, p);
+                        let mut failed_ips_guard = self.failed_addresses.failed_ips.lock().unwrap();
```

Comment:
imho don't unwrap if this fails, just skip appending and print warning

Reply:
in this case I could argue the unwrap is fine given it implied the mutex got poisoned, so there isn't much you could do anyway - but perhaps put an `expect` there instead for easier debugging